File: trunk/C#/RBD/RBD/DB/TableVersion.cs
// + TODO check

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Communication;

namespace RBD.DB
{
    public class TableVersion : BinarySerializable
    {
        public String getTableName()
        {
            return Name;
        }
        public void setTableName(String tableName)
        {
            Name = tableName;
        }
        public String Name { get; set; }

        public int getVersion()
        {
            return Version;
        }
        public void setVersion(int tableVersion)
        {
            Version = tableVersion;
        }
        public int getTableVersion() {
            return Version;
        }
        public void setTableVersion(int tableVersion) {
            Version = tableVersion;
        }
        public int Version { get; set; }

        public TableVersion(String name, int version)
        {
            Name = name;
            Version = version;
        }

        public TableVersion(DataInputStream dis) //throws IOException
        {
            FromBinary(dis);
        }

        public void FromBinary(DataInputStream dis) //throws IOException
        {
            Name = dis.ReadString();
            Version = dis.readInt();
        }

        public void ToBinary(DataOutputStream dos) //throws IOException
        {
            dos.WriteString(Name);
            dos.writeInt(Version);
        }

    }
}

File: tags/Release/SR-RBD/src/ddb/db/TableVersion.java
package ddb.db;

import java.io.IOException;

import ddb.BinarySerializable;
import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;

public class TableVersion implements BinarySerializable
{
	public TableVersion(String name, int version)
	{
		setTableName(name);
		setVersion(version);
	}
	
	public TableVersion(DataInputStream s) throws IOException
	{
		fromBinary(s);
	}
	
	public void setTableName(String tableName) {
		this.tableName = tableName;
	}
	public String getTableName() {
		return tableName;
	}

	public void setVersion(int version) {
		this.tableVersion = version;
	}

	public int getVersion() {
		return tableVersion;
	}
	
	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		tableName = s.readString();
		tableVersion = s.readInt();
	}
	
	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		s.writeString(tableName);
		s.writeInt(tableVersion);
	}

	private String tableName;
	private int tableVersion;
}
